addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRuleUnitTest.java

```diff
@@ -121,4 +129,225 @@ public void replaceMultipleElementsAndRemoveHeader() {
                         + "</comments>";
         assertThat(payload, is(expectedPayload));
     }
+
+    @Test
+    public void shouldScanOnlyIfRequestContentTypeIsXml() throws Exception {
+        // Given
+        String test = "/test";
+        this.nano.addHandler(
+                new NanoServerHandler(test) {
+                    @Override
+                    protected NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
+                        consumeBody(session);
+                        String response =
+                                "<foo>root:*:0:0:System Administrator:/var/root:/bin/sh</foo>";
+                        return newFixedLengthResponse(
+                                NanoHTTPD.Response.Status.OK, NanoHTTPD.MIME_PLAINTEXT, response);
+                    }
+                });
+
+        HttpMessage msg = this.getHttpMessage(test);
+        msg.setRequestBody("<?xml version=\"1.0\"?><comment><text>test</text></comment>");
+        msg.getRequestHeader().setHeader("Content-Type", "application/json");
```

Comment:
Is this an intentional mismatch?

Reply:
yes, it is. I have added a comment in the test to explain this mismatch.